pywbemtools/pywbemlistener/_cmd_listener.py

```diff
@@ -0,0 +1,1176 @@
+        if _config.VERBOSE_PROCESSES_ENABLED:
+            print("Start process: Cleaning up run process {} and status {}".
+                  format(child_pid, child_status))
+        try:
+            child_ps.terminate()
+            child_ps.wait()
+        except (IOError, OSError) as exc:
+            raise click.ClickException(
+                "Cannot clean up 'run' child process with PID {}: {}: {}".
+                format(child_pid, type(exc), exc))
+    return 1
+
+
+def run_exit_handler(start_pid, log_fp):
+    """
+    Exit handler that gets etablished for the 'run' command.
+
+    This exit handler signals a failed startup of the 'run' command
+    to the 'start' process, if it still exists. If the 'start'
+    process no longer exists, this means that the startup of the 'run'
+    command succeeded earlier, and it is now terminated by some means.
+
+    In addition, it closes the log_fp log file.
+    """
+    if _config.VERBOSE_PROCESSES_ENABLED:
+        print("Run process: Exit handler sends failure signal ({}) to start "
+              "process {}".
+              format(SIGNAL_RUN_STARTUP_FAILURE, start_pid))
+    try:
+        os.kill(start_pid, SIGNAL_RUN_STARTUP_FAILURE)  # Sends the signal
+    except OSError:
+        # The original start parent no longer exists -> the earlier startup
+        # succeeded.
+        pass
+
+    if log_fp:
+        print("Closing 'run' output log file at {}".format(datetime.now()))
+        log_fp.close()
+
+
+class DisplayDict(NocaseDict):
+    # pylint: disable=too-many-ancestors
+    """
+    Dictionary with a string representation that uses name=value format.
+    """
+
+    def __str__(self):
+        items = ['{}={!r}'.format(k, v) for k, v in self.items()]
+        return ' '.join(items)
+
+
+def format_indication(indication, host, indi_format=None):
+    """
+    Return a string that contains the indication formatted according to the
+    given format.
+    """
+    dt = datetime.now()
+    try:
+        dt = dt.astimezone()
+    except TypeError:
+        # Below Python 3.6, it cannot determine the local timezone, and its
+        # tzinfo argument is required.
+        pass
+    p_dict = DisplayDict()
+    for pn, p in indication.properties.items():
+        p_dict[pn] = p.value
+    format_kwargs = dict(
+        dt=dt,
+        dt_tzname=dt.tzname() or '',
+        h=host,
+        i=indication,
+        c=indication.classname,
+        p=p_dict,
+    )
+    if indi_format is None:
+        indi_format = DEFAULT_INDI_FORMAT
+    indi_str = indi_format.format(**format_kwargs)
+    return indi_str
+
+
+def show_help_format():
+    """
+    Display help for the format specification used with the --indi-format
+    option.
+    """
+    print("""
+Help for the format specification with option:  --indi-format FORMAT
+
+FORMAT is a new-style format string that can use the following keyword args:
+
+* 'dt' - datetime object of the time the listener received the indication, in
```

Comment:
It would be nice to have a counter per listener and be able to display it here as a variable.

Reply:
future feature.